Catalog item registration (#3656)
SkylineCommunications/dataminer-docs: user-guide/Cloud_Platform/Catalog/Tutorials/Tutorial_Register_Catalog_Version.md

---
uid: Tutorial_Register_Catalog_Version
---

# Registering a new version of a connector in the Catalog

> This tutorial demonstrates how to add a new version to a catalog item using the [Catalog API](xref:Register_Catalog_Item).  

Author: yes

> This tutorial demonstrates how to add a new version to a catalog item using the [Catalog API](xref:Register_Catalog_Item).  

Reviewer: ok

We will be registering our own version of the following example [connector](https://github.com/SkylineCommunications/SLC-C-Example_Rates-Custom), so go ahead and download the solution.

## Prerequisites

> - Organization key with permission "Register catalog items".

Reviewer: ![image](https://github.com/user-attachments/assets/bc6befff-3803-4f06-bdf9-bb612420b96e)

Don't see the key as a prerequisite anymore

> - Organization key with permission "Register catalog items".

Reviewer: Yeah, same as comment above I guess. You can leave it out if it's documented how to get it, only reason I'd keep a prerequisite is that a user can immediately see if they can follow this tutorial. If they are not an Owner or do not have a Key from an owner, then this tutorial becomes useless to them. Adding it as a prerequisite lets them quickly filter on that. I guess it's a choice to make, rather than an issue to address.

> - A registered catalog item, see [register a catalog item tutorial](xref:Tutorial_Register_Catalog_Item)

Author: ok

> - A registered catalog item, see [register a catalog item tutorial](xref:Tutorial_Register_Catalog_Item)

Reviewer: ok

## Overview

- [Step 1: Register catalog version URL](#step-1-register-catalog-version-url)
- [Step 2: Authentication header](#step-2-authentication-header)
- [Step 3: Register catalog version body](#step-3-register-catalog-version-body)
- [Step 4: Register](#step-4-register)

## Step 1: Register Catalog version URL

> In order to register our first version, we will use the catalog API Register version call.

Author: ok

> In order to register our first version, we will use the catalog API Register version call.

Reviewer: ok

Create a new Http request using the POST http method and url  <https://api.dataminer.services/api/key-catalog/v1-0/catalog/{catalogid}/version/register>

![Register version http url](~/user-guide/images/tutorial_catalog_registration_version_url.png)

>[!note]

> Make sure you use the same id as you used for the [catalog item registration](xref:Tutorial_Register_Catalog_Item).

Reviewer: ```suggestion
Make sure you use the correct Catalog ID in the URL. If you followed the tutorial to [register a catalog item](xref:Tutorial_Register_Catalog_Item), this was returned at the end.
```

Author: its actually generated manually by them at the start, the return in the response is just a confirmation (api no longer generates a GUID)

Reviewer: Ok, but if you word it like this you basically hard-link this tutorial to the one of the register item tutorial. It's perfectly possible they follow this tutorial with their own registered item, not the one from the other tutorial right?
I'd rephrase it slightly to make it clear it just has to be the correct ID

Author: aligned it here with the tutorial of github action

Reviewer: ok

## Step 2: Authentication Header

Add the previously obtained organization key in the **Ocp-Apim-Subscription-Key** header.

![Register version http header](~/user-guide/images/tutorial_catalog_registration_version_headers.png)

## Step 3: Register Catalog version Body

The body of the version registration request requires a file containing our connector, for this we will be using the 

> [Skyline.DataMiner.CICD.Tools.Packager NuGet](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.Packager#readme-body-tab) to create a .dmprotocol package from our solution.

Reviewer: ```suggestion
[Skyline.DataMiner.CICD.Tools.Packager NuGet](https://www.nuget.org/packages/Skyline.DataMiner.CICD.Tools.Packager#readme-body-tab) to create a `.dmprotocol` package from our solution.
```

Author: ok

Reviewer: doesn't appear in code block yet for me

Reviewer: ok

Navigate to [Example Rates connector](https://github.com/SkylineCommunications/SLC-C-Example_Rates-Custom) and download the solution.

Open a terminal window and run below command to install the packager.

```powershell

> dotnet tool install --global Skyline.DataMiner.CICD.Tools.Packager --version 2.0.3

Reviewer: Should this not be a listed prerequisite?
Also for the sake of this tutorial, do we need to have them create the package, can't we provide a readily compiled one?

Author: i feel its part of the experience to be able to generate a dmapp or dmprotocol. they got to know how to create that package otherwise they will never be able to upload their own . its not hard to do so , so i would keep it .

Added it as pre requisite

Reviewer: ok

```

> Run the following command to create a .dmprotocol package, adapt the directory paths as needed.

Reviewer: ```suggestion
Run the following command to create a `.dmprotocol` package, adapting the directory paths as needed.
```
Again, feels weird they need to run these for a register tutorial, why not give them a finished product to upload. Creating these packages is a different tutorial imo

Author: i prefer to keep it in, the "shared" part is the protocol solution from github which is what they normally start of with.
i would agree if it was complex and added a lot of steps but this isnt the case and we dont need to look for a location where we would make that pre package already available and possibly maintain it.

Reviewer: ok

Author: adapted dmprotocol syntax but not adapt to adapting cause that looks wrong. Its an instruction to user to adapt the paths as needed, not that the command will adapt any directories

Reviewer: Ok, will leave it @MariekeGO to correct our English sentences everywhere 😅

Reviewer 2: I agree with Alexander's suggestion, as the current text contains a [comma splice](https://www.grammarly.com/blog/comma-splice/). That means that either you need to split it up into two separate sentences, or you can change the second part to a subordinate clause. Changing it to a participle phrase as Alexander suggested is a perfectly valid option for this. In that case, "adapting" works as an instruction for what the user should do when they run the command, rather than something the command does.

```powershell
dataminer-package-create dmprotocol "C:\Tutorials\Catalog Registration\SLC-C-Example_Rates-Custom-1.0.1.X" --name catalog_registration_tutorial --output "C:\Tutorials\Catalog Registration\Packages"
```

> Add the file to the body as seen below and define versionNumber and versionDescription as well in the form-data

Reviewer: ```suggestion
Add the file to the body as seen below and define **versionNumber** and **versionDescription** as well in the **multipart/form-data**
```

Aligned it with the API docs: 
![image](https://github.com/user-attachments/assets/351d4f1c-fba7-4110-bafd-4681f62d091b)
http://localhost:8080/user-guide/Cloud_Platform/Catalog/Register_Catalog_Item.html

However I feel like here we shuffling steps a bit. Perhaps it's more clear to say: 'Add the file to the body as multipart/form-data' or something. Similar for the versionNumber and versionDescription. I would document these as explicit steps with bullet points so I can check them off as I do them mentally. Right now it's 3 things in 1 sentence.
The screenshot currently shows the 3 fields but the content type is set to Auto, yet here we specify the form data explicitly. I am personally confused on what is expected of me as a user here.

Author: i reformulated it. i originally didn't want to just do the same as it was described in the user guide with the bullet points. Changed it now.

Reviewer: ok

![Register version http body](~/user-guide/images/tutorial_catalog_registration_version_body.png)

## Step 4: Register

> Upon successful registration, you will be able to see the registered version on the catalog in the versions tab.

Reviewer: In the other tutorial you mention which status code they might receive in case of success, stay consistent and also add it here if you wish to add it there
Also capital C on catalog?

Author: ok, changed it

Reviewer: ok

> Upon successful registration, you will be able to see the registered version on the catalog in the versions tab.

Reviewer: ```suggestion
Upon successful registration, you will be able to see the registered version on the catalog in the **versions** tab.
```

Highlight the tab to indicate that versions isn't a 'word' but a thing they should look for (a tab). Can be bold or code block for example, just stay consistent :D

Author: ok, changed it

Reviewer: ok
